Get rid of CSV/TSV metrics (#1097)
iterative/dvc.org: content/docs/command-reference/metrics/index.md

## Synopsis

```usage

> usage: dvc metrics [-h] [-q | -v] {show,add,modify,remove,diff} ...

Reviewer: Oops, thanks!

positional arguments:

>   {show,add,modify,remove,diff}

Reviewer: Thanks also for updating this one. But actually we use this format specifically in the cmd ref:
```suggestion
COMMAND
```

Reviewer: Committing this one for consistency.

                        Use `dvc metrics CMD --help` to display command-
                        specific help.
    show                Print metrics, with optional formatting.
    add                 Mark a DVC-tracked file as a metric.
    modify              Modify metric default formatting.
    remove              Remove metric mark on a DVC-tracked file.

>     diff                Show changes in metrics between commits

Reviewer: Also, does it matter that the nav menu order is not the same as the output? If so, which one is right?

![image](https://user-images.githubusercontent.com/1477535/78108691-21b4e700-73b5-11ea-8380-212e7b33378d.png)

Reviewer: Cc @shcheklein

Author: I don't think it is important. But yes, it would be nice to make them the same.

Reviewer: OK, adjusted a little in 774edc74 (for #1110) and opened iterative/dvc/issues/3608.

optional arguments:
  -h, --help            show this help message and exit
  -q, --quiet           Be quiet.
  -v, --verbose         Be verbose
```

## Description

> In order to track metrics associated to machie learning experiments DVC has the

Reviewer: ```suggestion
In order to track metrics associated to machie learning experiments, DVC has the
```
I'll just commit this suggestion 🙂

> In order to track metrics associated to machie learning experiments, DVC has the
> ability to mark a certain stage <abbr>outputs</abbr> as files containing metrics
> to track. (See the `--metrics` option of `dvc run`.) Metrics are

Reviewer: > ...metrics to track.

Maybe we shouldn't mention the last part about tracking them if the plan is to encourage people to track them with Git? So:

```suggestion
In order to track metrics associated to machie learning experiments, DVC has the
ability to mark a certain stage <abbr>outputs</abbr> as files containing metrics.
Metrics are
```
> Needs rewrapping with following line.

Author: Good point. But let's keep it for now. We have already recommended using Git metrics.

Reviewer: OK, I guess when the --metrics option is removed this will be revisited. Resolving

project-specific floating-point values e.g. `AUC`, `ROC`, etc.

Supported file formats: JSON. Metrics can be organized in a tree hierarchy in a

> JSON file. DVC addresses the metrics by the tree path.

Reviewer: List of a single item seems strange.

Reviewer: "tree hierarchy"/"tree path" concepts need more explaining, maybe a quick example, or simplifying the text. Or even removing that part, since it's better explained in the metrics add/show cmd refs.

Reviewer: > List of a single item seems strange

p.s. I know YAML will be added but will it be soon? Prob better to have this as a sentence such as "Currently, DVC supports metric files in JSON format."

Author: Yes, it should be added soon. Let's keep it here for a week or two :)

DVC itself does not ascribe any specific meaning for these numbers. Usually

> these numbers are produced by the model training or model evaluation code and

Reviewer: ```suggestion
these numbers are produced by model training or evaluation code and
```

selectively extract data out of metric files:

```dvc
$ dvc metrics show --xpath AUC data/eval.json
      data/eval.json: {'AUC': 0.65115}
```

> The xpath filter can be saved for a metrics file:

Reviewer: ```suggestion
The `xpath` filter can be saved for a metrics file:
```

> The `xpath` filter can be saved for a metrics file:

Reviewer: Maybe even something like:
```suggestion
The `xpath` filter can be saved as the default way to display a metrics file:
```

Reviewer: Taking the freedom to commit this one too.
